staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -208,6 +208,11 @@ func (r ratchetingOptions) key(field string) ratchetingOptions {
 		return r
 	}
 
+	if r.nearestParentCorrelation == nil && (field == "metadata" || field == "apiVersion" || field == "kind") {
+		// Ratcheting is disabled for metadata fields.
```

Comment:
why? if the user put validation on labels or annotations, shouldn't that be able to ratchet?

Reply:
apiVersion and kind are blacklisted since the existing object is converted to the request version, so these fields are always equal.


I do agree that it was heavy handed to extend that restriction to `metadata`, I've removed that piece